### ISSUE TITLE
lldb 14.0.0 does not open on Windows

### ISSUE THREAD

**Issue by sr1canskhsia** [llvm/llvm-project#54589]
lldb 14.0.0 fails on startup and outputs
```
Python path configuration:
  PYTHONHOME = 'C:/Users/hwennborg/AppData/Local/Programs/Python/Python310'
  PYTHONPATH = (not set)
  program name = 'python'
  isolated = 0
  environment = 1
  user site = 1
  import site = 1
  sys._base_executable = 'C:\\Program Files\\LLVM\\bin\\lldb.exe'
  sys.base_prefix = 'C:/Users/hwennborg/AppData/Local/Programs/Python/Python310'
  sys.base_exec_prefix = 'C:/Users/hwennborg/AppData/Local/Programs/Python/Python310'
  sys.platlibdir = 'lib'
  sys.executable = 'C:\\Program Files\\LLVM\\bin\\lldb.exe'
  sys.prefix = 'C:/Users/hwennborg/AppData/Local/Programs/Python/Python310'
  sys.exec_prefix = 'C:/Users/hwennborg/AppData/Local/Programs/Python/Python310'
  sys.path = [
    'C:\\Program Files\\Python\\python310.zip',
    'C:/Users/hwennborg/AppData/Local/Programs/Python/Python310\\DLLs',
    'C:/Users/hwennborg/AppData/Local/Programs/Python/Python310\\lib',
    'C:\\Program Files\\LLVM\\bin',
  ]
Fatal Python error: init_fs_encoding: failed to get the Python codec of the filesystem encoding
Python runtime state: core initialized
ModuleNotFoundError: No module named 'encodings'

Current thread 0x0000251c (most recent call first):
  <no Python frame>
```

My OS is Windows 10 and I already have Python 3.10.4 installed under C:\Program Files\Python\

A workaround is to extract python310.zip from https://www.python.org/ftp/python/3.10.4/python-3.10.4-embed-amd64.zip and put it under C:\Program Files\Python\, but why would I need to do this if I already had Python installed in the first place?

**Comment by razaqq** [comment]
same here

**Comment by trondhe** [comment]
I mailed a tiny bit with Hans Wennborg, as its his name that appears as a username of the python configuration. He said that the windows installer gets build in his pc. It seems to be saved inside the lldb binary or in a dynamic lib in the build process, as I can't find in while grepping for it.

This is an excerpt from his mail
```
I only uploaded the installer like 24 hours ago, so maybe more bug
reports will come :-/

You can see the script I used here:
https://discourse.llvm.org/t/14-0-0-final-has-been-tagged/61153/2

One thing that changed in this release, and is maybe related, is that I
switched from Python 3.6 (I think) to 3.10. To do that I had to tweak
some CMake flags. The new script is linked above, and the old one is here:

https://lists.llvm.org/pipermail/release-testers/attachments/20211004/0be53ef5/attachment.obj

But both of them were setting PYTHON_HOME, so that hasn't changed at least.
```

I sadly havent had time to debug this further, if there is some cmake script that incorporates info that it shouldnt when packaging the installer.

Building lldb from source works fine.

**Comment by bertiqwerty** [comment]
Same here. I will try to build from source.

**Comment by Aatroxs** [comment]
Same here. Also for 14.0.1,with Python 3.10.4.

**Comment by kouhe3** [comment]
same here, LLVM 14.0.3 , Python 3.10.4

**Comment by TerryBLK** [comment]
same here, LLVM 14.03, Python 3.10.4

**Comment by Emilgardis** [comment]
Building from source will make it work on your system, the current build script is https://discourse.llvm.org/t/14-0-4-final-has-been-tagged/62750/3

I don't think more reports of people having the issue will help here, rather the cause has to be found.

A quick look, this could be something: https://github.com/llvm/llvm-project/blob/5560b9e88423dbe044f4e82d4dc3d4b6dbd0e678/lldb/cmake/modules/LLDBConfig.cmake#L146-L154

**Comment by Emilgardis** [comment]
I think actually this is why, the option has been renamed from `LLDB_RELOCATABLE_PYTHON` to `LLDB_EMBED_PYTHON_HOME`, but in the docs it's still called `LLDB_RELOCATABLE_PYTHON` [here](https://github.com/llvm/llvm-project/blob/09c2b7c35af8c4bad39f03e9f60df8bd07323028/lldb/docs/resources/build.rst#windows-1) and in the buildscript, it's used as like that

**Comment by Emilgardis** [comment]
ping @zmodem

**Comment by zmodem** [comment]
Thanks for digging into this! https://reviews.llvm.org/D74727 seems like the relevant change.

Sounds like we just need to add `-DLLDB_EMBED_PYTHON_HOME=OFF` to the build.

I made a new build of 14.0.4 with that set. Can some try it out? https://drive.google.com/file/d/1i1JlhTxhiuIv1m8VIfyN0LS9rDw02Z2L/view?usp=sharing

**Comment by Emilgardis** [comment]
Works!

The build instructions should be updated to reflect this change

**Comment by sr1canskhsia** [comment]
@zmodem I can confirm that this new build works on my machine

**Comment by zmodem** [comment]
Thanks everyone for confirming, I've updated the release build script, so the next releases will be built with the fix.

**Comment by EugeneZelenko** [comment]
@zmodem: I think will be good idea to backport fix to 14,x branch.

**Comment by zmodem** [comment]
The fix (so far) is in how we invoke the build, and that's not on the branch itself. There is another 14 release planned, 14.0.5, I suppose that will be sometime next week, and I'll make sure to build that with the updated script.

If someone wants to fix the cmake files themselves, that would be great and something we could backport, but I'm not qualified to touch lldb's cmake files myself.

**Comment by sr1canskhsia** [comment]
Fixed in 14.0.5. Closing
